Production_scripts/Costs-02-district_nursing.R

```diff
@@ -0,0 +1,368 @@
+  outfile %>%
+  full_join(dn_raw_costs_contacts, by = c("HB2019", "year"))
+
+
+
+# recode NA pop_proportion with 1
+data <-
+  data %>%
+  mutate(pop_proportion = replace_na(pop_proportion, 1))
+
+
+
+## total net cost ##
+data <-
+  data %>%
+  mutate(cost_total_net = ((cost * 1000) / (number_of_contacts / pop_proportion))
+         ) %>%
+  # sort by HB2019
+  arrange(HB2019) %>%
+  # keep only records with cost
+  filter(!is.na(cost_total_net))
+
+
+
+# ----------------------------------------------------------
+
+
+## Fix incomplete submisssions ##
+# If a Partnership has abnormally low contacts this will affect the cost so use the previous year until we have a complete submission
+
+# explore the trends
+
+data <-
+  data %>%
+  group_by(Board_Name) %>%
+  mutate(max_contacts = max(number_of_contacts)) %>%
+  mutate(pct_of_max = number_of_contacts / max_contacts * 100)
+
+
+
+# plot #
+ggplot(data = data, aes(x = year, y = pct_of_max, group = Board_Name)) +
+  geom_line(aes(color = Board_Name)) +
+  labs(color = "NHS Board", x = "Year")
+
+
+
+## upflift any 'copied' costs ##
+data <-
+  data %>%
+  mutate(uplift = 0) %>%
+  mutate(tempyear1 = case_when(Board_Name == "NHS Highland" & year == "1617"                ~ "1819",
+                               Board_Name == "NHS Tayside" & year == "1617"                 ~ "1718",
+                               Board_Name == "NHS Forth Valley" & year == "1819"            ~ "1920",
+                               Board_Name == "NHS Greater Glasgow & Clyde" & year == "1819" ~ "1920"),
+         tempyear2 = case_when(Board_Name == "NHS Highland" & year == "1617"                ~ "1920")
+         )%>%
+  mutate(uplift1 = case_when(Board_Name == "NHS Highland" & tempyear1 == "1819"                  ~ 2,
+                             Board_Name == "NHS Tayside" & tempyear1 == "1718"                   ~ 1,
+                             Board_Name == "NHS Forth Valley" & tempyear1 == "1920"              ~ 1,
+                             Board_Name == "NHS Greater Glasgow & Clyde" & tempyear1 == "1920"   ~ 1),
+         uplift2 = case_when(Board_Name == "NHS Highland" & tempyear2 == "1920"                  ~ 3)
+  )
+
+
+# data - wide to long
+data <-
+  data %>%
+  pivot_longer(
+    c("year", contains("tempyear")),
+    values_to = "year",
+    names_to = NULL
+  ) %>%
+  pivot_longer(
+    contains("uplift"),
+    values_to = "uplift",
+    names_to = NULL
+  ) %>%
+  # cost total net
+  mutate(cost_total_net = cost_total_net * (1.01 * exp(uplift)))
+
+
+
+## Add in years by copying the most recent year we have ##
+latest_year <- 1920
+data <-
+  bind_rows(
+    data,
+    map_df(1:5, ~
+             data %>%
+             filter(year == latest_year) %>%
+             mutate(
+               cost_total_net = cost_total_net * (1.01)^.x,
+               year = convert_year_to_fyyear(as.numeric(convert_fyyear_to_year(year)) + .x)
+             ))
+  )
+
+
+
+data <-
+  data %>%
+  ungroup() %>%
```

Comment:
confused why this is here - best practice to do it after a group and move further up

Reply:
@Jennit07 think it was James that put this ungroup() in? but moved up to same %>% as `group_by()`